docs/proposals/20230524-support-nodepool-with-host-network-mode.md

```diff
@@ -0,0 +1,158 @@
+
+It can be seen below that kube-proxy/flannel/coredns components are occupying a significant amount of CPU and memory.
+
+```
+$ k top po -A --sort-by=memory
+NAMESPACE      NAME                                       CPU(cores)   MEMORY(bytes)
+kube-system    kube-apiserver-master1                     127m         350Mi
+kube-system    etcd-master1                               29m          81Mi
+kube-system    kube-controller-manager-master1            43m          49Mi
+cert-manager   cert-manager-cainjector-74bfccdfdf-4kvf7   3m           30Mi
+kube-system    kube-proxy-fttg6                           1m           22Mi
+kube-flannel   kube-flannel-ds-cd9f6                      7m           21Mi
+cert-manager   cert-manager-b4d6fd99b-bj9jm               2m           17Mi
+kube-system    coredns-7d89d9b6b8-vhqj7                   4m           16Mi
+kube-system    coredns-7d89d9b6b8-fn9c4                   3m           16Mi
+kube-system    kube-scheduler-master1                     6m           16Mi
+kube-system    metrics-server-66dd897cc4-zl2sq            8m           14Mi
+kube-system    metrics-server-59d8dc4bc-54v59             6m           13Mi
+kube-flannel   kube-flannel-ds-cs9dp                      9m           11Mi
+cert-manager   cert-manager-webhook-65b766b5f8-z9k6c      5m           11Mi
+kube-system    kube-proxy-x2v2r                           3m           10Mi
+default        nginx-pod                                  0m           3Mi
+```
+
+### Goals
+
+Implement a new type of NodePool that enables Pods on the nodes to use the host network while not using services, CoreDNS, and other components.
+
+### Non-Goals/Future Work
+
+Refactor `yurtadm join` and so on.
+
+## Proposal
+
+### Definition
+
+* NodePool with host network mode
+  * A new NodePool type where Pods will only use host networking mode, suitable for lightweight and minimally demanding edge nodes.
+
+### Architecture
+
+Here is a example for NodePool in host network mode:
+
+```
+apiVersion: apps.openyurt.io/v1beta1
+kind: NodePool
+metadata:
+  name: beijing
+spec:
+  type: Cloud
+  hostNetwork: true
+```
+
+The work we need to do is as follows:
+
+* **Step1**: First, before adding a node to the NodePool, you need to modify the kubelet configuration of the node. This part of the work can be done by yurtadm join.
+
+/var/lib/kubelet/kubeadm-flags.env
+
+`—network-plugin=cni`
+
+`—cni-conf-dir=/etc/cni/net.d`
+
+For example:
+
+```
+KUBELET_KUBEADM_ARGS="--network-plugin=cni —cni-conf-dir=/etc/cni/net.d --pod-infra-container-image=registry.cn-hangzhou.aliyuncs.com/google_containers/pause:3.5"
```

Comment:
set kubelet --network-plugin="" will make node work in host network mode.

Reply:
--network-plugin string                                    The name of the network plugin to be invoked for various events in kubelet/pod lifecycle. This docker-specific flag only works when container-runtime is set to docker. (DEPRECATED: will be removed along with dockershim.)
network-plugin only work with docker, so it's not a good way.

Reply:
Thank you for your advice.

Yes, when --network-plugin="", we will use docker network bridge.

![image](https://github.com/openyurtio/openyurt/assets/53330071/a58968c4-fe26-489b-8c53-06de219de0b0)


![image](https://github.com/openyurtio/openyurt/assets/53330071/7ab45a84-1a72-43be-8f2c-08fac989c889)

network-plugin only work with docker, That's a problem, because that means it's not a universal approach.